ray-operator/controllers/ray/utils/httpproxy_httpclient.go

```diff
@@ -5,29 +5,43 @@ import (
 	"io"
 	"net/http"
 	"time"
+
+	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 type RayHttpProxyClientInterface interface {
 	InitClient()
+	WithKubernetesPodProxy(podNamespace, podName string, port int)
 	CheckHealth() error
 	SetHostIp(hostIp string, port int)
 }
 
-func GetRayHttpProxyClient() RayHttpProxyClientInterface {
-	return &RayHttpProxyClient{}
+func GetRayHttpProxyClientFunc(mgr ctrl.Manager) func() RayHttpProxyClientInterface {
+	return func() RayHttpProxyClientInterface {
+		return &RayHttpProxyClient{
+			mgr: mgr,
+		}
+	}
 }
 
 type RayHttpProxyClient struct {
-	client       http.Client
+	client       *http.Client
 	httpProxyURL string
+
+	mgr ctrl.Manager
 }
 
 func (r *RayHttpProxyClient) InitClient() {
-	r.client = http.Client{
+	r.client = &http.Client{
 		Timeout: 20 * time.Millisecond,
 	}
 }
 
+func (r *RayHttpProxyClient) WithKubernetesPodProxy(podNamespace, podName string, port int) {
```

Comment:
The function initializes the client. Perhaps we could update InitClient instead of creating a new function?